components/dot_menu/index.ts

```diff
@@ -65,8 +68,21 @@ function mapStateToProps(state: GlobalState, ownProps: Props) {
     const currentTeam = getCurrentTeam(state) || {};
     const currentTeamUrl = `${getSiteURL()}/${currentTeam.name}`;
 
+    const rootId = post.root_id || post.id;
+    const systemMessage = isSystemMessage(post);
+    let threadReplyCount = 0;
+    let isFollowingThread = false;
+    let threadId = rootId;
+
+    if (rootId && !systemMessage) {
```

Comment:
This should be behind feature flag

Reply:
> Overall looks good to me! I'm just seeing one crash-error when you open RHS @mentions. Here's a patch for a few suggestions regarding that and a couple others: [MM-32784.patch.zip](https://github.com/mattermost/mattermost-webapp/files/6481652/MM-32784.patch.zip) (probably needs a few tweaks—e.g. CenterChannel dot-menu reply)

Thanks, I didn't see that!
 
> When I added `getThreadOrSynthetic`, I went back/forth if the signature should be `(state, threadId)` or `(state, post)`. Reason for the latter: I think search-result posts aren't added to `PostsState['posts']`.

You are right, search results throws, I updated the function signature to `(state, post)`,
actually it should be `(state, root)`, else we are going to have incorrect data e.g `reply_count` is 0 for replies.

We could `throw` if there is a `post.root_id` to enforce it, a bit HC though, what do you think?